src/client/Microsoft.Identity.Client/AppConfig/ApplicationConfiguration.cs

```diff
@@ -160,6 +160,8 @@ public X509Certificate2 ClientCredentialCertificate
                 return null;
             }
         }
+
+        public IDictionary<string, string> CacheKeyComponents { get; internal set; }

```

Comment:
There is no advantage of using a interface here. It will lead to upcasting later. Use SortedDictionary.